src/Symfony/Component/Cache/Traits/MemcachedTrait.php

```diff
@@ -198,6 +200,12 @@ protected function doSave(array $values, $lifetime)
             $lifetime += time();
         }
 
+        if ($this->isTextProtocol) {
+            $encoded_values = array();
+            array_walk($values, function ($value, $key) use (&$encoded_values) { $encoded_values[rawurlencode($key)] = $value; });
```

Comment:
I generally don't like array_walk(), too much references there. Should be done with a foreach instead.
The temp car should use camel case coding name also.